### PR TITLE
Refactor: move end-to-end test infrastructure to @grafana/toolkit

### PR DESCRIPTION
**Pull request by ryantxu**
In #17967 we would like to support puppeteer test on plugins in circleci

This moves the setup files to toolkit so they can be reused.

### CODE REVIEW

**Comment by ryantxu** [comment]
@dprokop -- I'm stumped on what to try changing next.   any ideas?

```
$ yarn e2e-tests                                                                                     
yarn run v1.17.3
$ jest --runInBand --config=jest.config.e2e.js
 FAIL  public/e2e-test/scenarios/smoke.test.ts
  ● Test suite failed to run

    TypeError: Cannot read property 'fromAriaLabel' of undefined

      17 |   url: '/login',
      18 |   pageObjects: {
    > 19 |     username: new InputPageObject(Selector.fromAriaLabel('Username input field')),
         |                                            ^
      20 |     password: new InputPageObject(Selector.fromAriaLabel('Password input field')),
      21 |     submit: new ClickablePageObject(Selector.fromAriaLabel('Login button')),
      22 |   },

      at Object.<anonymous> (packages/grafana-toolkit/src/e2e/start/loginPage.ts:19:44)
      at Object.<anonymous> (packages/grafana-toolkit/src/e2e/login.ts:4:1)
      at Object.<anonymous> (packages/grafana-toolkit/src/e2e/index.ts:5:1)
      at Object.<anonymous> (packages/grafana-toolkit/src/index.ts:1:1)
      at Object.<anonymous> (public/e2e-test/scenarios/smoke.test.ts:3:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.913s
Ran all test suites.
error Command failed with exit code 1.
```

**Comment by ryantxu** [comment]
@davkal -- This does not aim to change anything, simply put the classes in `@grafana/toolkit`

The changes in formatting were automatic by prettier tooling, and since `@grafana/toolkit` has stricter types we needed to add some optional flags and checks

**Comment by ryantxu** [comment]
@davkal -- can you check this again?

I now have both master and this branch failing with:

```
    console.log packages/grafana-toolkit/src/e2e/pageObjects.ts:59
      Trying to click on: [aria-label="Link to rendered image"]

  ● Login scenario, create test data source, dashboard, panel, and export scenario › should pass

    Error [ERR_UNHANDLED_ERROR]: Unhandled error. (Error:
    The screenshot:[./public/e2e-test/screenShots/theOutput/smoke-test-scenario.png] taken during the test differs by:[52571] pixels from the expected.

    Compare the output from expected:[./public/e2e-test/screenShots/theTruth] with outcome:[./public/e2e-test/screenShots/theOutput]

      If the difference between expected and outcome is NOT acceptable then do the following:
        - Check the code for changes that causes this difference, fix that and retry.

      If the difference between expected and outcome is acceptable then do the following:
        - Replace the expected image with the outcome and retry.

      68 |         msg += '\n    - Replace the expected image with the outcome and retry.';
      69 |         msg += '\n';
    > 70 |         throw new Error(msg);
         |               ^
      71 |       }
      72 |
      73 |       resolve();

      at Object.<anonymous>.exports.PNG.doneReading (packages/grafana-toolkit/src/e2e/images.ts:70:15)
      at Object.<anonymous>.exports.PNG.<anonymous> (node_modules/pngjs/lib/png.js:37:10)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ParserAsync._complete (node_modules/pngjs
```

File: packages/grafana-toolkit/package.json [hunk 1]
```diff
@@ -78,5 +83,8 @@
   "devDependencies": {
     "@types/glob": "^7.1.1",
     "@types/prettier": "^1.16.4"
+  },
+  "_moduleAliases": {
```
Comment (dprokop) [inline packages/grafana-toolkit/package.json]: What is this needed for?

Comment (ryantxu) [reply packages/grafana-toolkit/package.json]: no idea -- it was in the existing 🤷‍♂

Comment (davkal) [reply packages/grafana-toolkit/package.json]: the npm package `puppeteer` split a while ago into `puppeteer-core` and friends. This was needed for backward compat of packages that referred to it as `puppeteer`. Not sure if this is still needed.

**Review by davkal — APPROVED** [review]
I'm happy with this PR even though font rendering differences between OSX/Linux make it impossibnle to get a full pass on OSX as long as the truth pic comes from Linux.

N.B. I've tried the threshold settings, but had to dial it up to .75 where it became meaningless to detect small changes like added characters by mistake.

**Pull request merged**